gateway/ln-gateway/src/lib.rs

```diff
@@ -251,30 +310,22 @@ impl Gatewayd {
                                 // Blocks until the connection to the lightning node breaks
                                 info!("Established HTLC stream");
 
-                                // Re-create gateway
-                                let gateway = Gateway::new(
-                                    ln_client.clone(),
-                                    client_builder.clone(),
-                                    self.fees.clone().unwrap_or(GatewayFee(DEFAULT_FEES)).0,
-                                    database.clone(),
-                                    self.api_addr.clone(),
-                                    clients.clone(),
-                                    scid_to_federation.clone(),
-                                    tg.clone(),
-                                    self.num_route_hints,
-                                )
-                                .await.expect("Failed to created Gateway");
-
-                                info!("Successfully created Gateway");
-
-                                let task_group = run_webserver(self.password.clone(), self.listen, gateway)
-                                    .await
-                                    .expect("Failed to start webserver");
-                                info!("Successfully started webserver");
-
-                                Gateway::handle_htlc_stream(stream, ln_client, handle.clone(), scid_to_federation.clone(), clients.clone()).await;
-                                warn!("HTLC Stream Lightning connection broken. Stopping webserver...");
-                                task_group.shutdown();
+                                let (lightning_public_key, lightning_alias) =
+                                Self::fetch_lightning_node_info(ln_client.clone()).await.expect("Failed to retrieve Lightning info after establishing a connection");
```

Comment:
Perhaps retry this and other operations below? But this could be another PR

Reply:
The likelihood of this failing is hopefully small since just above it established the intercepted HTLC stream, but I added the retry anyway.